### PR TITLE
Fixes listview shrinkwrap and hidden semantics node gets updated incorrectly.

### LINKED ISSUE #84719: Drop down menu does not scroll using a screen reader
## Steps to Reproduce

Using a screen reader run the code below:
1. Open the menu by tapping on the `DropdownButton`.
2. Swipe to navigate to through the list
3. When reaching the end of the visible list focus will leave the list instead of scrolling to the next item

```dart
import 'package:flutter/material.dart';

main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  MyAppState createState() => MyAppState();
}

class MyAppState extends State<MyApp> with WidgetsBindingObserver {
  @override
  Widget build(context) => MaterialApp(title: 'Semantics bug demo app', home: HomePage());
}

class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  int selected = 0;
  @override
  Widget build(context) => Scaffold(
        appBar: AppBar(title: Text("testing")),
        body: Center(
          child: DropdownButton(
            items: List.generate(100, (index) => DropdownMenuItem<int>(child: Text("Item #$index "), value: index)),
            value: selected,
            onChanged: (value) => setState(() => selected = value),
          ),
        ),
      );
}
```

**Expected results:** 
As a screen reader user, I expect the semantic focus to go to the next item of the list of items until it reaches the end.

**Actual results:** 
However, when the focus reaches the end of the visible items in the scrollable region it announces "Dismiss", ignoring most of the items in the list. There is no indication to the screenreader that there are more items or that the list is scrollable.

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/84719

2 issues:

1. RenderShrinkWrapViewport does not annotate the semantics clip rect correctly.
2. RenderIndexSemantics should not be semantics boundary because it is usually used under scrollable which will contribute semantics property hidden to the children.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/rendering/viewport.dart [hunk 1]
```diff
@@ -1920,6 +1920,14 @@ class RenderShrinkWrappingViewport extends RenderViewportBase<SliverLogicalConta
...
     _shrinkWrapExtent = 0.0;
     _hasVisualOverflow = false;
+    switch (cacheExtentStyle) {
+      case CacheExtentStyle.pixel:
+        _calculatedCacheExtent = cacheExtent;
+        break;
+      case CacheExtentStyle.viewport:
+        _calculatedCacheExtent = mainAxisExtent * _cacheExtent;
+        break;
+    }
```
Comment (dnfield) [inline packages/flutter/lib/src/rendering/viewport.dart:1930]: I probably missed this when I changed the way cache extent could be specified :(

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 2]
```diff
@@ -5056,7 +5056,6 @@ class RenderIndexedSemantics extends RenderProxyBox {
   @override
   void describeSemanticsConfiguration(SemanticsConfiguration config) {
     super.describeSemanticsConfiguration(config);
-    config.isSemanticBoundary = true;
```
Comment (dnfield) [inline packages/flutter/lib/src/rendering/proxy_box.dart:5059]: How does this not break things?

Comment (chunhtai) [reply packages/flutter/lib/src/rendering/proxy_box.dart:5059]: the scrollable will be the semantics boundary. In its assemble semantics node, it still create a separate semantics node for every of its children. So the isSemanticsBoundary in RenderIndexSemantics is just performance improvement

**Review by dnfield — APPROVED** [review]
LGTM

**Pull request merged**